libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -97,10 +97,26 @@ func (out *otelConsumer) logsPublish(_ context.Context, batch publisher.Batch) e
 		pcommonEvent.CopyTo(logRecord.Body().SetEmptyMap())
 	}
 
-	if err := out.logsConsumer.ConsumeLogs(context.TODO(), pLogs); err != nil {
-		return fmt.Errorf("error otel log consumer: %w", err)
+	err := out.logsConsumer.ConsumeLogs(ctx, pLogs)
+	if err != nil {
+		// Permanent errors shouldn't be retried. This tipically means
+		// the data cannot be serialized by the exporter that is attached
+		// to the pipeline or when the destination refuses the data because
+		// it cannot decode it. Retrying in this case is useless.
+		//
+		// See https://github.com/open-telemetry/opentelemetry-collector/blob/1c47d89/receiver/doc.go#L23-L40
+		if consumererror.IsPermanent(err) {
```

Comment:
Does the definition of `IsPermanent` align with ours, and in cases where we don't, do we do the same things?

Edge cases I can think of at the moment:
- We recover from 413 Entity Too Large responses by splitting up batches. https://github.com/elastic/beats/pull/34911. This possibly needs to live in the Elasticsearch exporter but could also live here.
- We consider 409 Conflict to be permanent, which hopefully upstream does too based on the definition, but it can be quite command and even normal when the [fingerprint](https://www.elastic.co/guide/en/beats/filebeat/current/fingerprint.html) processor is used to add a document level _id for deduplication.

Reply:
Yes, I expected they'd be handled separately just wanted to raise them first.